potodo/_po_file.py

```diff
@@ -0,0 +1,70 @@
+import polib
+
+from typing import Mapping
+from pathlib import Path
+
+
+class PoFileStats:
+    """Class for each `.po` file containing all the necessary information about its progress
+    """
+
+    def __init__(self, path: Path):
+        """Initializes the class with all the correct information
+        """
+        self.path: Path = path
+        self.filename: str = path.name
+        self.pofile: polib.POFile = polib.pofile(self.path)
+        self.directory: str = self.path.parent.name
+
+        self.fuzzy_entries: list = self.pofile.fuzzy_entries()
+        self.fuzzy_nb: int = len(self.fuzzy_entries)
+
+        self.percent_translated: int = self.pofile.percent_translated()
+
+        self.translated_entries: list = self.pofile.translated_entries()
+        self.translated_nb: int = len(self.translated_entries)
+
+        self.untranslated_entries: list = self.pofile.untranslated_entries()
+        self.untranslated_nb: int = len(self.untranslated_entries)
+
+        self.obsolete_entries: list = self.pofile.obsolete_entries()
+        self.obsolete_nb: int = len(self.pofile.obsolete_entries())
+
+        self.po_file_size = len(self.pofile) - self.obsolete_nb
+
+        self.filename_dir: str = self.directory + "/" + self.filename
+
+    def __str__(self) -> str:
+        return (
+            f"Filename: {self.filename}\n"
+            f"Fuzzy Entries: {self.fuzzy_entries}\n"
+            f"Percent Translated: {self.percent_translated}\n"
+            f"Translated Entries: {self.translated_entries}\n"
+            f"Untranslated Entries: {self.untranslated_entries}"
+        )
+
+    def __lt__(self, other: "PoFileStats") -> bool:
+        """When two PoFiles are compared, their filenames are compared.
+        """
+        return self.filename < other.filename
+
+
+def get_po_files_from_repo(repo_path: str) -> Mapping[str, list]:
```

Comment:
Is this ok @JulienPalard ?

Reply:
We can do better with `Mapping[str, List[PoFileStats]]`, that way when you do `get_po_files_from_repo(".")["foo"][0]` type checkers can infer it's a `PoFileStats` instance, an continue checking, like if you do `get_po_files_from_repo(".")["foo"][0].replace("foo", "bar")` you should get a nice "PoFileStats has no replace method".

But wait, we can do better, do we really care it's a List? Maybe just tell it has `__getitem__` `__iter__`, `__len__` and so on by just telling it's a `Sequence`.

Reply:
So, `Mapping[str, Sequence[PoFileStats]]`.

Reply:
Niiice, thanks ! 
```suggestion
def get_po_files_from_repo(repo_path: str) -> Mapping[str, Sequence[PoFileStats]]:
```